apps/web/components/modals/server-settings-modal.tsx

```
@@ -142,6 +183,52 @@ export function ServerSettingsModal({ open, onClose, server, isOwner, channels =
+              {/* Server Icon */}
+              {isOwner && (
+                <div className="space-y-2">
+                  <Label className="text-xs font-semibold uppercase tracking-wider" style={{ color: 'var(--theme-text-secondary)' }}>
+                    Server Icon
+                  </Label>
+                  <div className="flex items-center gap-4">
+                    <div
+                      onClick={() => iconFileRef.current?.click()}
+                      className="w-20 h-20 rounded-full border-2 border-dashed flex items-center justify-center cursor-pointer hover:border-white/50 transition-colors overflow-hidden relative"
+                      style={{ borderColor: 'var(--theme-text-faint)' }}
+                    >
```

Biome (2.4.4)

[error] Enforce to have the onClick mouse event with the onKeyUp, the onKeyDown, or the onKeyPress keyboard event.

(lint/a11y/useKeyWithClickEvents)